kong/clustering/compat/init.lua

```diff
@@ -312,6 +312,61 @@ do
 end
 
 
+local compatible_checkers = {
+  { 3002000000, --[[ 3.2.0.0 ]]
+    function(config_table, dp_version, log_suffix)
+      local config_plugins = config_table["plugins"]
+      if not config_plugins then
+        return nil
+      end
+
+      local has_update
+      for _, plugin in ipairs(config_plugins) do
+        if plugin["instance_name"] ~= nil then
+          plugin["instance_name"] = nil
+          has_update = true
+        end
+      end
+
+      if has_update then
+        ngx_log(ngx_WARN, _log_prefix, "Kong Gateway v" .. KONG_VERSION ..
+                " contains configuration 'plugin.instance_name'",
+                " which is incompatible with dataplane version " .. dp_version,
+                " and will be removed.", log_suffix)
+      end
+
+      return has_update
+    end
+  },
+
+  { 3001000000, --[[ 3.1.0.0 ]]
+    function(config_table, dp_version, log_suffix)
+      local config_upstreams = config_table["upstreams"]
+      if not config_upstreams then
+        return nil
+      end
+
+      local has_update
+      for _, t in ipairs(config_upstreams) do
+        if t["use_srv_name"] ~= nil then
+          t["use_srv_name"] = nil
+          has_update = true
+        end
+      end
+
+      if has_update then
+        ngx_log(ngx_WARN, _log_prefix, "Kong Gateway v" .. KONG_VERSION ..
+                " contains configuration 'upstream.use_srv_name'",
+                " which is incompatible with dataplane version " .. dp_version,
+                " and will be removed.", log_suffix)
+      end
+
+      return has_update
+    end
+  },
+}
+
+
```

Comment:
I think It will be nice if you can add this table to separate lua file like this https://github.com/Kong/kong/blob/master/kong/clustering/compat/removed_fields.lua

Reply:
done.